admin/controllers/Staff_groups.php

```diff
@@ -162,10 +162,21 @@ public function edit() {
 	}
 
 	private function _saveStaffGroup() {
+                $input_post=$this->input->post();
+                 
+                    $permissions=$this->Permissions_model->getPermissionsByIds();
+                    
+                     
+                    foreach($input_post['permissions'] as $permission_id=>$permission_access){
+                        $input_post['permissions'][$permissions[$permission_id]['name']]= $input_post['permissions'][$permission_id];
+                        //$group_permissions[$permission_info['permission_id']]=$group_permissions[$permission_name];
+                        log_message('info',print_r( $input_post['permissions'],true).'@@@@@@@@@@@ permissions assigned to the staff groups of the user');
```

Comment:
I think this can be a better way just needs a little more work, also modifying xss_clean changes how uri's are formed and sometimes throw errors where the files cannot be located because of the incorrect path.